mm2src/coins/rpc_command/get_estimated_fees.rs

```diff
@@ -0,0 +1,413 @@
+use crate::eth::{EthCoin, FeePerGasEstimated};
+use crate::AsyncMutex;
+use crate::{from_ctx, lp_coinfind, MarketCoinOps, MmCoinEnum, NumConversError};
+use common::executor::{SpawnFuture, Timer};
+use common::log::debug;
+use common::{HttpStatusCode, StatusCode};
+use futures::channel::mpsc;
+use futures::{select, Future, StreamExt};
+use futures_util::FutureExt;
+use mm2_core::mm_ctx::MmArc;
+use mm2_err_handle::mm_error::MmError;
+use std::collections::HashSet;
+use std::pin::Pin;
+use std::sync::Arc;
+
+/// RPCs to start/stop gas price estimator and get estimated base and priority fee per gas
+
+const FEE_ESTIMATOR_REFRESH_INTERVAL: u32 = 15; // in sec
+const FEE_ESTIMATOR_NAME: &str = "eth_fee_estimator_loop";
+const MAX_CONCURRENT_STOP_REQUESTS: usize = 10;
+
+pub(crate) type FeeEstimatorStopListener = mpsc::Receiver<String>;
+pub(crate) type FeeEstimatorStopHandle = mpsc::Sender<String>;
+
+/// Gas fee estimator running loop state
+enum FeeEstimatorState {
+    Starting,
+    Running,
+    Stopping,
+    Stopped,
+}
+
+impl Default for FeeEstimatorState {
+    fn default() -> Self { Self::Stopped }
+}
+
+// Errors
+
+#[derive(Debug, Display, Serialize, SerializeErrorType)]
+#[serde(tag = "error_type", content = "error_data")]
+pub enum FeeEstimatorError {
+    #[display(fmt = "Coin not activated or not a EVM coin")]
+    CoinNotFoundOrSupported,
```

Comment:
I believe we should seperate these errors to be more as they are quite different scenarios (Not activated / Not an EVM coin)

Reply:
done